### ISSUE TITLE
SAM build container with environment variables

### ISSUE THREAD

**Issue by forzagreen** [aws/aws-sam-cli#2144]
### Description

When building lambda functions in Docker container ( `sam build --use-container` ), we can pass in a Docker network (with `--docker-network`), but you can't pass environment variables.

It would be very useful to be able to provide environment variables.

### Use Case

Python lambda function using packages from a private registry. You won't have to hardcode your credentials in `requirements.txt`

### Proposal

Add a new parameter to pass environment variables. Something like:
```sh
sam build --use-container --docker-env VAR1=value1 --docker-env VAR2=value2
```
or:
```sh
sam build --use-container --docker-env-file dev.env
```

**Comment by willfish** [comment]
@awsjeffg What's the pm-review process? Looking at the container/CLI implementation it seems like this is pretty straightforward to achieve from a code perspective?

**Comment by morefaey** [comment]
Thanks @willfish for your comment. The pm-review label is for us to track and review feature requests and rank the ones that we have to work on. you are right that this straightforward to achieve, and we are currently reviewing it to see when it gets implemented.

**Comment by staenker** [comment]
So, is that backlog grooming meeting still going on? If so, I personally like a good beard and a good beard definitely deserves good grooming. So if the beard is the issue, you have my full understanding and emotional support. 
If not, could you maybe consider pushing this showstopper issue for using sam with containers for anything more than a hello world to a higher rank?

**Comment by qingchm** [comment]
@staenker Hi there yes we are looking to support it, sorry for the wait but I will try to work on this as soon as possible! But please understand that there are other tasks with high ranks as well! Again apologies on the wait!

**Comment by nickswiss** [comment]
I feel like this may be too specific to only env vars. Secrets are also a highly used feature in docker builds and I think something along the lines of `--docker-build-flags` might work even better, which would cover both use cases.

**Comment by qingchm** [comment]
@forzagreen Sorry for all the wait, just a heads up this implementation is done and under review now, so you should be able to install private registries by passing in an environment variable json file as a build option very soon (if I'm correctly understanding your use case). Again thanks for being patient with our SAM CLI ;) hopefully this update will make you satisfied!

**Comment by qingchm** [comment]
Released this feature with our v1.20.0, closing the issue now, let me know if you have any further questions!
